src/passes/transform/clock_gate.cpp

```diff
@@ -0,0 +1,124 @@
+#include "coreir/passes/transform/clock_gate.h"
+
+std::string CoreIR::Passes::ClockGate::ID = "clock_gate";
+
+namespace {
+struct CEInfo {
+  bool can_replace = false;
+  CoreIR::Instance *reg = nullptr;
+  CoreIR::Instance *mux = nullptr;
+  int mux_in_port = 0;
+  CEInfo() : can_replace(false) {}
+  CEInfo(CoreIR::Instance *reg, CoreIR::Instance *mux, bool mux_in_port)
+      : can_replace(true), reg(reg), mux(mux), mux_in_port(mux_in_port) {}
+};
```

Comment:
Not sure on what's standard (CC @rsetaluri), but any reason this is a struct vs class?

Reply:
I think a struct is fine here since I am just using it as a custom return type for this local function.